src/System.Console/src/System/ConsolePal.Windows.cs

```diff
@@ -218,6 +329,12 @@ private static ConsoleColor ColorAttributeToConsoleColor(Interop.mincore.Color c
             return (ConsoleColor)c;
         }
 
+        private static Interop.mincore.CONSOLE_SCREEN_BUFFER_INFO GetBufferInfo()
+        {
+            bool unused;
+            return GetBufferInfo(true, out unused);
+        }
+
```

Comment:
> How about on Unix creating a child process that opens a chr device (e.g. /dev/null) and dups it onto fd 1; then we 
should detect that console out is a terminal and proceed as if we're not redirected.

WIth <3 days experience on Unix, I am not sure I fully grasp this :). I will take a closer look. Also, I was looking at some of the Console tests especially for TermInfo where we have removed the test from CI system but still added some tests which we can run manually. We can add some like that :)

> Even if we don't do that, though, we should still test what we can, that we throw PlatformNotSupported in the appropriate cases, that we don't throw exceptions in other cases, etc
Yes, I will add these scenarios..